sys/net/crosslayer/pktbuf/pktbuf.c

```diff
@@ -20,21 +20,24 @@
 #include <string.h>
 
 #include "mutex.h"
+#include "pkt.h"
 #include "pktbuf.h"
 
-#if PKTBUF_SIZE < 128
-typedef uint8_t _pktsize_t;
-#elif PKTBUF_SIZE < 65536
-typedef uint16_t _pktsize_t;
-#else
-typedef size_t _pktsize_t;
+#ifdef DEVELHELP
+#include "od.h"
 #endif
 
-typedef struct __attribute__((packed)) _packet_t {
-    volatile struct _packet_t *next;
+/**
+ * @brief   Data type to represent describe chunks in packet buffer.
+ */
+typedef struct __attribute__((packed)) _pkt_internal_t {
+    struct _pkt_internal_t *next;
     uint8_t processing;
-    _pktsize_t size;
-} _packet_t;
+    pkt_hlist_t *header_ptr;
```

Comment:
The struct is still packed, can you maybe test this on the platforms you are concerned this might occur? The unittests for pktbuf have a test for alignment problems (`test_pktbuf_insert_packed_struct()`). In general I find it difficult to move the processing counter (and think int would be overkill), since the current implementation expects the pkt_t struct always directly before the data block (so you can find it very fast, if you only have the data pointer for e. g. copy).